BUG: Pullmode: using KustomizationRefs with ContinuousWithDriftDetection causes addon controller to crash

# Problem Description

When KustomizationRefs are used with ContinuousWithDriftDetection, addon controller crashes 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x2162b35]

goroutine 106 [running]:
github.com/projectsveltos/libsveltos/lib/deployer.deployResourceSummaryInstance({0x32a8d88, 0xc000842b90}, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0x4db16a0, 0x0, 0x0}, ...)
        /go/pkg/mod/github.com/projectsveltos/libsveltos@v1.2.0/lib/deployer/drift_detection_utils.go:96 +0xf5
github.com/projectsveltos/libsveltos/lib/deployer.DeployResourceSummaryInCluster({0x32a8d88, 0xc000842b90}, {0x0, 0x0}, {{0x2e2b054?, 0x1?}, {0xc007300d80?, _}}, {0xc003e7d290, 0x24}, ...)
        /go/pkg/mod/github.com/projectsveltos/libsveltos@v1.2.0/lib/deployer/drift_detection_utils.go:75 +0x445
github.com/projectsveltos/addon-controller/controllers.deployResourceSummaryWithKustomizeResources({0x32a8d88, 0xc000842b90}, {0xc003e7d290, 0x24}, {0xc003e7d2b7, 0x24}, 0xc00617e388, {0xc003e7d2de, 0x7}, {0xc008f91b30, ...}, ...)
        /workspace/controllers/handlers_kustomize.go:1021 +0x3c5
github.com/projectsveltos/addon-controller/controllers.handleKustomizeResourceSummaryDeployment({0x32a8d88, 0xc000842b90}, 0xc00617e388, {0xc003e7d290, 0x24}, {0xc003e7d2b7, 0x24}, {0xc003e7d2de, 0x7}, {0xc008f91b30, ...}, ...)
        /workspace/controllers/handlers_kustomize.go:988 +0x14a
github.com/projectsveltos/addon-controller/controllers.handleDriftDetectionManagerDeploymentForKustomize({0x32a8d88, 0xc000842b90}, 0xc00617e388, {0xc003e7d290, 0x24}, {0xc003e7d2b7, 0x24}, {0xc003e7d2de, 0x7}, 0x1, ...)
        /workspace/controllers/handlers_kustomize.go:969 +0x20f
github.com/projectsveltos/addon-controller/controllers.deployKustomizeRefs({0x32a8d88, 0xc000842b90}, {0x32c83f0, 0xc00050d050}, {0xc003e7d290, 0x24}, {0xc003e7d2b7, 0x24}, {0xc003e7d2e8, 0x3a}, ...)
        /workspace/controllers/handlers_kustomize.go:123 +0x36a
github.com/projectsveltos/addon-controller/controllers.genericDeploy({0x32a8d88, 0xc000842b90}, {0x32c83f0, 0xc00050d050}, {0xc003e7d290, 0x24}, {0xc003e7d2b7, 0x24}, {0xc003e7d2e8, 0x3a}, ...)
        /workspace/controllers/clustersummary_deployer.go:320 +0x1a4
github.com/projectsveltos/libsveltos/lib/deployer.processRequests({0x32a8d88, 0xc000842b90}, 0xc000051950, 0xa, {{0x32b2ec8?, 0xc0008719c0?}, 0x0?})
        /go/pkg/mod/github.com/projectsveltos/libsveltos@v1.2.0/lib/deployer/worker.go:175 +0x4d3
created by github.com/projectsveltos/libsveltos/lib/deployer.(*deployer).startWorkloadWorkers in goroutine 1
```

--- a/controllers/handlers_kustomize.go
+++ b/controllers/handlers_kustomize.go
@@ -230,7 +230,7 @@ func processKustomizeDeployment(ctx context.Context, remoteRestConfig *rest.Conf
 	}
 
 	err = handleKustomizeResourceSummaryDeployment(ctx, clusterSummary, clusterSummary.Spec.ClusterNamespace,
-		clusterSummary.Spec.ClusterName, clusterSummary.Spec.ClusterType, remoteDeployed, logger)
+		clusterSummary.Spec.ClusterName, clusterSummary.Spec.ClusterType, remoteDeployed, isPullMode, logger)
 	if err != nil {
 		return err
 	}
@@ -967,7 +967,7 @@ func handleDriftDetectionManagerDeploymentForKustomize(ctx context.Context, clus
 		// un-needed reconciliation (Sveltos is updating those resources so we don't want drift-detection to think
 		// a configuration drift is happening)
 		err = handleKustomizeResourceSummaryDeployment(ctx, clusterSummary, clusterNamespace, clusterName,
-			clusterType, []libsveltosv1beta1.Resource{}, logger)
+			clusterType, []libsveltosv1beta1.Resource{}, isPullMode, logger)
 		if err != nil {
 			logger.V(logs.LogInfo).Error(err, "failed to remove ResourceSummary.")
 			return err
@@ -981,9 +981,15 @@ func handleDriftDetectionManagerDeploymentForKustomize(ctx context.Context, clus
 // ResourceSummary in the managed cluster
 func handleKustomizeResourceSummaryDeployment(ctx context.Context, clusterSummary *configv1beta1.ClusterSummary,
 	clusterNamespace, clusterName string, clusterType libsveltosv1beta1.ClusterType,
-	remoteDeployed []libsveltosv1beta1.Resource, logger logr.Logger) error {
+	remoteDeployed []libsveltosv1beta1.Resource, isPullMode bool, logger logr.Logger) error {
 
 	if clusterSummary.Spec.ClusterProfileSpec.SyncMode == configv1beta1.SyncModeContinuousWithDriftDetection {
+		// If SveltosCLuster is in pull mode, do nothing. A knob will notify agent about DriftDetection being set
+		// and agent will manage the ResourceSummary life cycle in the managed cluster
+		if isPullMode {
+			return nil
+		}
+
 		// deploy ResourceSummary
 		err := deployResourceSummaryWithKustomizeResources(ctx, clusterNamespace, clusterName, clusterSummary,
 			clusterType, remoteDeployed, logger)

